File: data/scripts/white_to_alpha.js
// Aseprite
// Copyright (C) 2015-2016 by David Capello

var col = app.pixelColor
var img = app.activeImage

for (y=0; y<img.height; ++y) {
  for (x=0; x<img.width; ++x) {
    var c = img.getPixel(x, y)
    var v = (col.rgbaR(c)+
             col.rgbaG(c)+
             col.rgbaB(c))/3

    img.putPixel(x, y,
                 col.rgba(col.rgbaR(c),
                          col.rgbaG(c),
                          col.rgbaB(c),
                          255-v))
  }
}

File: data/scripts/white_to_alpha.lua
-- Aseprite
-- Copyright (C) 2015-2018 by David Capello

local pc = app.pixelColor
local img = app.activeImage

for y=0,img.height-1 do
   for x=0,img.width-1 do
    local c = img:getPixel(x, y)
    local r = pc.rgbaR(c)
    local g = pc.rgbaG(c)
    local b = pc.rgbaB(c)
    local a = pc.rgbaA(c)
    if a > 0 then a = 255 - (r+g+b)/3 end
    img:putPixel(x, y, pc.rgba(r, g, b, a))
  end
end
